Add a script to update the region/geolocation files

absidue opened the pull request:
# Add a script to update the region/geolocation files

<!-- Thanks for sending a pull request! Make sure to follow the contributing guidelines. -->
<!-- Important note, we may remove your pull request if you do not use this provided PR template correctly. -->

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [x] Bugfix
- [x] Feature Implementation

## Related issue
close #3441

## Description
Currently FreeTube lists a bunch geolocations for trending that YouTube doesn't support and is also missing some that YouTube does support, this is because the locations were taken from a repository that had location information translated into various different languages.

This pull request adds a script that extracts the geolocations from YouTube in as many of the active languages in FreeTube as possible, that way we have an accurate list and also have them translated into various different languages.

For some languages I've added manual mappings, like French which is `fr-FR` in FreeTube but `fr` on YouTube.

The script currently only collects geolocation translations for whitelisted/active languages in FreeTube, it doesn't for any of the other languages.

Here is the script output:
```
yarn get-regions
yarn run v1.22.19
$ node _scripts/getRegions.mjs
Active FreeTube languages that aren't available on YouTube:
[ 'es-MX', 'es_AR', 'he', 'pt-BR' ]
YouTube languages that don't have an equivalent active FreeTube language:
[
  'af',     'am',    'as',    'az',
  'be',     'bn',    'bs',    'en-IN',
  'es-419', 'es-US', 'fa',    'fil',
  'fr-CA',  'gu',    'hi',    'hy',
  'iw',     'ka',    'kk',    'km',
  'kn',     'ky',    'lo',    'lv',
  'mk',     'ml',    'mn',    'mr',
  'ms',     'my',    'ne',    'or',
  'pa',     'si',    'sq',    'sr-Latn',
  'sw',     'ta',    'te',    'th',
  'ur',     'uz',    'zh-HK', 'zu'
]
Done in 2.70s.
```

## Screenshots
![three-dots-menu](https://github.com/FreeTubeApp/FreeTube/assets/48293849/d88a8ffb-5ac3-4729-8aa8-2cb51cd19fea)

## Testing <!-- for code that is not small enough to be easily understandable -->
Test trending with various locations, none of them should error like they did before.

PikachuEXE commented:
What do you mean by "Test trending with **various locations**" ?_?
On of the options in settings? (I haven't used that part ever)

ChunkyProgrammer commented on _scripts/getRegions.mjs:
```diff
@@ -0,0 +1,128 @@
+
+// Scrape the location menu in various languages and write files to the file system
+
+rmSync(`${STATIC_DIRECTORY}/geolocations`, { recursive: true })
```
I feel like we shouldn't remove the directory and should just overwrite matching files. This will allow us to add other files that are supported by FreeTube but not by YouTube (ex: notice how some files were removed without a new version)

absidue replied:
Some of the ones that were removed were for languages that we don't even have in FreeTube, not even as inactive ones like `ee` or `zh` (we have `zh-CN` and `zh-TW`, which both have YouTube equivalents).

The main issue with maintaining our own files, is that when they inevitably become outdated (YouTube adding and removing regions), you will be able to select non-existent regions (brings us back to the issue this pull request is meant to solve) and won't be able to select new ones until they are translated.

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (changes requested):
Messaging should be changed.

![Capture](https://github.com/FreeTubeApp/FreeTube/assets/73130443/08b3fa79-ffb8-48b2-8c1d-579a0cd870a2)


Not sure if we need to do anything about this but if u select some languages the alphabetic order of the list gets a bit messed up.


https://github.com/FreeTubeApp/FreeTube/assets/73130443/8b4abdab-c41e-4212-a395-a50db46b11ef


https://github.com/FreeTubeApp/FreeTube/assets/73130443/c39ea4dc-9e45-4ad3-8e6a-6d7c652eee36

absidue commented:
Added language sensitive sorting, the only language that changed was `de-DE`, I even intentionally changed the order of the `cs` language but it went back to putting `Chile` after `Hong Kong`, so in the first video the sorting seems to be correct.

absidue commented:
Removed the extra sentence and added sorting for the languages, any that looks incorrectly sorted now, is correct unless the standard javascript Intl.Collator API is broken, which i highly doubt.
